pkg/controller/jobsyncstrategy.go

```diff
@@ -71,3 +71,10 @@ type JobSyncStrategy interface {
 	// do not want to support reprocessing should return nil.
 	GetLastJobSuccess(owner metav1.Object) *time.Time
 }
+
+// CheckBeforeDeprovision should be implemented by a strategy to have the sync loop check
+// whether a particular owner resource is ready for deprovisioning when a DeletionTimestamp is detected.
+type CheckBeforeDeprovision interface {
```

Comment:
We should use the same terminology here that we use elsewhere in `JobSync`. Specifically, the inverse of process is undo. `JobSync` does not have a concept of deprovision. Either `CheckBeforeDeprovision` should be `CheckBeforeUndo` or we should change the undo terminology to something else in `JobSync`.

Reply:
Ack